lib/plugins/threaded-signing/plugin.c

```diff
@@ -215,32 +262,27 @@ threaded_openssl_sign_hash(sv_threaded_plugin_t *self, const signature_info_t *s
   if (signature_info->hash_size != self->hash_size) goto catch_error;
 
   // Copy the |hash| ready for signing.
-  memcpy(self->hash_to_sign, signature_info->hash, signature_info->hash_size);
+  memcpy(
+      self->input_buffer[self->input_buffer_idx], signature_info->hash, signature_info->hash_size);
+  self->input_buffer_idx++;
 
   status = SV_OK;
-  self->plugin_state = THREADED_SIGNING_HAS_HASH_TO_SIGN;
 
 catch_error:
-  if (status == SV_UNKNOWN_FAILURE) {
+  if (status != SV_OK) {
     // Failed in memory allocation. Free all memory and set status to SV_MEMORY.
+    if (status == SV_UNKNOWN_FAILURE) status = SV_MEMORY;
```

Comment:
I think you should remove this line and instead make sure ```status``` is set right before each ```goto catch_error;```